Install language files when installing packages

mlocati opened the pull request:
What about automatically downloading language files when a package is installed?

When installing a package via the dashboard, this option is controlled by the new `concrete.i18n.auto_install_package_languages` configuration key.

When installing a package via the CLI command, users may specify if they want to download the language files or not with the new `--get-languages` option (if this option is not specified, we'll use the `concrete.i18n.auto_install_package_languages` configuration key).

The language files that will be downloaded are the ones used in the site (that is, the languages of the multilingual sections and the languages of the site interface).

aembler commented:
This is great in theory - but I'm wondering if we couldn't just simply inform users that there's this nice multilingual update page in the Dashboard where they could download the translations they want as well? I'm also concerned about package name collisions, to the point where I think we would only want to do this if a user had connected their site to the community and had a matching license for the package attached to their project page. We could still leave the other interface in the Dashboard for downloading any package translation that matched any handle.

mlocati commented:
Automatically install language files is much easier and faster than redirecting users to the language update file, but I do agree that someone wouldn't want to automatically install package language files (for instance for custom packages not in PRB).
An alternative to the link to the Update Languages page we could add a new property to the package controllers (defaulting to true): if it's false the automatic language installation is inhibited...

mlocati commented:
I updated this pull request: every package can say if it's in the concrete5 marketplace (that implies that its translations are managed in translate.concrete5.org).
By default we assume this is true. To say this is false, simply add this line to the custom package `controller.php`:
```php
    protected $isMarketplacePackage = false;
```

This flag is now also used in the `/dashboard/system/basics/multilingual/update` marketplace package, to avoid installing wrong language files for packages that are not in the marketplace.

If this approach is OK, and if this PR gets merged, we have to make sure that packages submitted to PRB doesn't have `$isMarketplacePackage` set to `false` (in order to avoid human work, we could add a new rule to the PRB linter - cc @JohntheFish ).

aembler commented:
I've done some thinking about this and I'm fine with it automatically downloading package translations (rather than redirecting to an update screen.) But I don't like adding the new property to the controller. We already have a secure way of knowing definitively whether a package is part of the markteplace – we connect the site to the community and download that way. That's how we handle package updates, after all. Let's change this method to verify the packages the user has as part of their site from those that are connected to their project page. 

`Concrete\Core\Marketplace\Marketplace::getAvailableMarketplaceItems(false);`

will return a list of all add-ons that are connected to this site from the marketplace. Let's use that to determine which languages to download.

mlocati commented:
:+1:

mlocati commented:
I updated this pull request.

Now, language files are installed if the following criteria are met:
- installing packages via the dashboard:
  - the new `concrete.i18n.auto_install_package_languages` configuration key is set to `true` (the default)
  - the package being installed is associated to the project

When installing packages via the `c5:package:install` CLI command, we'll use the value of a new `--languages` option:
  - if it's `yes`: language files are installed
  - if it's `no`: language files are not installed
  - if it's `auto` (the default): language files are installed provided that
    - the canonical URL is set (required by `Marketplace::getAvailableMarketplaceItems`)
    - the package being installed is associated to the project

PS: I also fixed some warnings (ref. #4723)

aembler commented:
Thanks!
